docker-compose.yaml

```diff
@@ -0,0 +1,24 @@
+version: "3.7"
```

Comment:
Is there a way we can avoid using a .yaml file? Could we have all the logic in the DockerFile? It seems like were running the DockerFile is used to run the server rather than init/run the database. Since npm already has all these scripts, we should just start the database and then start the server to connect to it. 

We should move the init and starting database logic to the DockerFile so we don't need the .yaml file. Let me know if you need help doing this :)

Reply:
From my understanding of Docker, Dockerfile is used for building the server image, while docker-compose.yaml specifies the services used and also the docker network that connected the services together. In the case where we bundled both express and psql into the Dockerfile - what would happen if we deploy the container img onto something like ECS? Would both of them ran on the same container instance?

Also are there a reason for not using docker-compose? I'm genuinely asking here, since I haven't came across materials that compare the usage of .yaml and Dockerfile

Reply:
But why do we need to to use the Dockerfile to run the server when npm does that for us already through its scripts? It seems like we only need the docker-compose file here to run the db and then the server will connect to it. 

We could run both of them together on an ECS instance but wouldn't it be easier to run the server on ELB and then the db on RDS? That way we have easier access to both of them rather than SSH into the instance and then modifying everything there. That's why I think that regardless of using the DockerFile or docker-compose, we are only using it mainly for local development. Unless you think otherwise. I just think that it is an easier process.

Reply:
@kendricchung I feel like it would probably be better if I clarify my choices here - we are probably mis-understanding each other in some parts.
1. The Dockerfile is solely for building the image of express server, and not the db. The way I use Dockerfile before was to use it to build the web server image, and deploy this image into ECS. This ECS would usually be connected to a RDS, like you said. 
2. When I talked about running them onto the same container, it's because I thought you are suggesting to use Dockerfile to build both express and psgl into one image, and I was saying I also don't prefer that. But I think we both agree on this aspect.
3. The docker-compose file is to run the express and psql on local containers, for local development. They are not really used for deployment.
4. From your response, do you only want the postgres to be deployed (with docker), and not express?

Reply:
@pkbui I can see that we are on the same page for the docker-compose. I just think that we should not have the server run on the Docker File since we can have it run on ELB in the future. I know that ELB is much easier to use since we don't need to SSH into the instance itself. We can configure the instance on the AWS website itself. I think that we both have similar ideas but different service uses. Lets talk about this more on Wed.